### PR TITLE
build: implementing ory kratos and ory hydra

### PR DESCRIPTION
**Pull request by LandonPattison**
### What
Implemented Ory Kratos

### Related issue(s) and discussion

- Fixes #[6865]

### CODE REVIEW

File: kratos/auth.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      - POSTGRES_USER=kratos
+      - POSTGRES_PASSWORD=secret
+      - POSTGRES_DB=kratos
+    networks:
+      - intranet
+
+  kratos-migrate:
+    image: oryd/kratos:v0.9.0-alpha.3
+    links:
+      - postgres-kratos:postgres-kratos
```
Comment (alexgarel) [inline kratos/auth.yml:16]: I'm not sure why we need that and if we need that.

File: kratos/auth.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  postgres-kratos:
+    image: postgres:9.6
+    environment:
+      - POSTGRES_USER=kratos
+      - POSTGRES_PASSWORD=secret
+      - POSTGRES_DB=kratos
+    networks:
+      - intranet
+
+  kratos-migrate:
```
Comment (alexgarel) [inline kratos/auth.yml:13]: We don't need this service here.

File: kratos/auth.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    volumes:
+      - type: bind
+        source: ./kratos
+        target: /etc/config/kratos
+    command: -c /etc/config/kratos/kratos.yml migrate sql -e --yes
+
+  kratos:
+    image: oryd/kratos:v0.9.0-alpha.3
+    links:
+      - postgres-kratos:postgres-kratos
```
Comment (alexgarel) [inline kratos/auth.yml:30]: I think you don't need this.

File: kratos/auth.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+version: '3.7'
+
+services:
+  postgres-kratos:
+    image: postgres:9.6
+    environment:
+      - POSTGRES_USER=kratos
+      - POSTGRES_PASSWORD=secret
+      - POSTGRES_DB=kratos
```
Comment (alexgarel) [inline kratos/auth.yml:9]: Add a volume for data.

File: kratos/auth.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      - intranet
+    ports:
+      - '4455:3000'
+    restart: on-failure
+
+  mailslurper:
+    image: oryd/mailslurper:latest-smtps
+    ports:
+      - '4436:4436'
+      - '4437:4437'
```
Comment (alexgarel) [inline kratos/auth.yml:59]:
```suggestion
    ports:
      - '127.0.0.1:4436:4436'
      - '127.0.0.1:4437:4437'
```

File: kratos/identity.schema.json [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+              },
+              "last": {
+                "title": "Last Name",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "required": [
+          "email"
```
Comment (alexgarel) [inline kratos/identity.schema.json:44]: + userid

File: kratos/auth.yml [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  kratos:
+    image: oryd/kratos:v0.9.0-alpha.3
+    links:
+      - postgres-kratos:postgres-kratos
+    environment:
+      - DSN=postgres://kratos:secret@postgres-kratos:5432/kratos?sslmode=disable&max_conns=20&max_idle_conns=4
+    ports:
+      - '4433:4433'
+      - '4434:4434'
+    volumes:
```
Comment (alexgarel) [inline kratos/auth.yml:36]: you can verify it's ok by running:

```
docker-compose run --rm kratos bash
$ ls -l /etc/config/kratos
```

and you should see your config files. (you may cat them to verify)

File: kratos/auth.yml (hunk 4 above)
Comment (LandonPattison) [reply kratos/auth.yml:9]: will add closer to production don't want to save users quite yet created issue for later, https://github.com/openfoodfacts/openfoodfacts-server/issues/6907

File: kratos/kratos.yml [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+dsn: memory
+
+serve:
+  public:
+    base_url: http://127.0.0.1:4433/
+    cors:
+      enabled: true
+  admin:
+    base_url: http://kratos:4434/
```
[the remark this answers is not in the document]
Comment (LandonPattison) [reply kratos/kratos.yml]: fixed with last commit

File: kratos/kratos.yml [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+              hook: session
+
+log:
+  level: debug
+  format: text
+  leak_sensitive_values: true
+
+secrets:
+  cookie:
+    - PLEASE-CHANGE-ME-I-AM-VERY-INSECURE
```
[the remark this answers is not in the document]
Comment (LandonPattison) [reply kratos/kratos.yml:64]: yes will set through env var, made an issue for now with links to the docs, https://github.com/openfoodfacts/openfoodfacts-server/issues/6908

File: kratos/kratos.yml [hunk 10]
```diff
@@ -0,0 +1,84 @@
+version: v0.10.0
+
+dsn: memory
```
[the remark this answers is not in the document]
Comment (LandonPattison) [reply kratos/kratos.yml:3]: we can use memory for testing but will want to set it to postgres one eventually yes I'm not so sure how this works so will create an issue for now, https://github.com/openfoodfacts/openfoodfacts-server/issues/6909

File: kratos/auth.yml (hunk 3 above)
Comment (LandonPattison) [reply kratos/auth.yml]: I don't believe so

**Comment by alexgarel** [comment]
Hi @LandonPattison I see you put it back in draft.
We could have merged it into main branch.
When you start a new feature, if you create a new branch it will help making different PRs thus better tracking progress.

More over from now on, it would be better if you open branches directly from this git repository, you have the permissions for it.

**Comment by LandonPattison** [comment]
@alexgarel Im a little confused on last comment. Are you saying just commit to the OryKratos branch then do PRs to the main branch? If that is correct I misunderstood I thought I was supposed to create a OryKratos branch then do PRs to the OryKratos branch, then it would eventually be merged with the main branch.

**Comment by LandonPattison** [comment]
Im gonna merge this fork to OryKratos branch and for future PRs make draft pr to main branch

**Pull request merged**